modules/test/playwright/tests/frontend-data-set-views-web/itemActions.spec.ts

```diff
@@ -9,68 +9,941 @@ import {featureFlagsTest} from '../../fixtures/featureFlagsTest';
+				confirmationMessage: LINK_ITEM_ACTION_CONFIRMATION_MESSAGE,
+				icon: 'link',
+				name: LINK_ITEM_ACTION_NAME,
+				type: 'link',
+				url: 'https://www.liferay.com/',
+			});
+		});
+
+		const siteInfo =
+			await test.step('Go home and create a new site', async () => {
+				await page.goto('/');
+
+				const newSite = await fdsFragmentPage.createSite(SITE_NAME);
+
+				const pageLayout = await fdsFragmentPage.createPage({
+					siteId: newSite.id,
+					title: PAGE_NAME,
+				});
+
+				await page.reload();
+
+				return {layout: pageLayout, site: newSite};
+			});
+
+		await test.step('Edit page', async () => {
+			await fdsFragmentPage.editPage({...siteInfo});
+		});
+
+		await test.step('Search for "Data Set" fragment', async () => {
+			await fdsFragmentPage.searchFragmentOrWidget('Data Set');
+		});
+
+		await test.step('Drag "Data Set" fragment & Drop into the page editor w/ keyboard', async () => {
+			await fdsFragmentPage.dragAndDropFragment(
+				'Data Set Add Data Set Mark Data Set as Favorite'
+			);
+		});
+
+		await test.step('Select empty Data Set fragment', async () => {
+			await page
+				.getByText('Select a data set view. Beta')
+				.first()
+				.click();
+		});
+
+		await test.step('Open Data Set View Selector', async () => {
+			await page
+				.getByRole('button', {name: 'Select Data Set View'})
+				.click();
+		});
+
+		await test.step('Select Data Set View', async () => {
+			await expect(page.getByRole('dialog')).toBeVisible();
+
+			await expect(
+				page.getByRole('heading', {name: 'Select'})
+			).toBeVisible();
+
+			await page
+				.frameLocator('iframe[title="Select"]')
+				.locator('li')
+				.filter({hasText: DATASET_VIEW_NAME})
+				.first()
+				.click();
+
+			await page
+				.frameLocator('iframe[title="Select"]')
+				.getByRole('button', {name: 'Save'})
+				.click();
+		});
+
+		await test.step('Publish page with Data Set View', async () => {
+			await fdsFragmentPage.publishPage();
+
+			await fdsFragmentPage.goToPage({...siteInfo});
+
+			await expect(page.locator('.data-set-wrapper')).toBeVisible();
+		});
+
+		await test.step('Item actions are present in table row', async () => {
+			const dialogPromise = page
+				.waitForEvent('dialog')
+				.then(async (dialog) => {
+					await dialog.accept();
+
+					return dialog.message();
+				});
+
+			await page.locator('.dnd-td.item-actions').first().waitFor();
+
+			const tableRow = await page.locator('.dnd-td.item-actions').first();
+
+			await expect(tableRow.getByRole('link')).toBeVisible();
+
+			await tableRow.getByRole('link').click();
+
+			const confirmationMessage = await dialogPromise;
+
+			expect(confirmationMessage).toBe(
+				LINK_ITEM_ACTION_CONFIRMATION_MESSAGE
+			);
+
+			await page.waitForURL('https://www.liferay.com');
+
+			await expect(page.url()).toContain('https://www.liferay.com');
+		});
+
+		await test.step('Delete Data Set and site', async () => {
+			await fdsFragmentPage.goto();
```

Comment:
Need to go back to Liferay to be able to delete the site. Otherwise, getting a `connection closed` error.